app/models/concerns/checkout.rb

```diff
@@ -85,4 +114,24 @@ def are_deals_quantity_valid?
       end
     end
 
+    def can_be_cancelled?
+      if line_items.includes(:deal).all? { |line_item| line_item.deal.end_at - Time.current > ENV['minutes_before_expiration_when_deal_can_be_cancelled'].to_i.minutes }
```

Comment:
check if this can be done via rails query

Reply:
`ENV['minutes_before_expiration_when_deal_can_be_cancelled'].to_i.minutes` move this to a cosntant in constant.rb